code/modules/paperwork/paper.dm

```diff
@@ -462,9 +461,26 @@
 /**Stamp the paper with the  specified values.
  * stamper_name: what is stamped. Or what comes after the sentence "This paper has been stamped "
 */
-/obj/item/paper/proc/apply_custom_stamp(var/image/stamp, var/stamper_name)
+/obj/item/paper/proc/apply_custom_stamp(stamp_icon, stamper_name, offset_x, offset_y)
+
+	if(!stamp_icon || !check_state_in_icon("stamp", stamp_icon))
+		return
+
+	var/image/stamp = overlay_image(stamp_icon, "stamp", COLOR_WHITE, RESET_COLOR)
+
+	if(isnull(offset_x))
+		stamp.pixel_x = rand(-2, 2)
+	else if(offset_x != 0)
+		stamp.pixel_x = offset_x
+
+	if(isnull(offset_y))
+		stamp.pixel_y = rand(-2, 2)
+	else if(offset_y != 0)
+		stamp.pixel_y = offset_y
+
+	// TODO: alpha mask against paper.
```

Comment:
can we not just use `blend_mode = BLEND_INSET_OVERLAY` on the stamp image? for reference
> [BLEND_INSET_OVERLAY](https://www.byond.com/docs/ref/index.html#/atom/var/blend_mode) overlays the icon, but masks it by the image being drawn on. This is pretty much not at all useful directly on the map, but can be very useful for an overlay for an atom that uses KEEP_TOGETHER (see [appearance_flags](https://www.byond.com/docs/ref/info.html#/atom/var/appearance_flags)), or for the [layering filter](https://www.byond.com/docs/ref/info.html#/{notes}/filters/layer).